### PR TITLE
Add logging around suspicious FormEntryActivity code

### PR DESCRIPTION
**Pull request by seadowg**
Adds logging that attempts to give us more info around https://github.com/getodk/collect/issues/5241.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by seadowg** [comment]
@grzesiek2010 @lognaturel feel free to add other logs in for things you're curious about here.

File: collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java [hunk 1]
```diff
@@ -253,6 +255,7 @@ protected void onCleared() {
     }
 
     private void updateIndex() {
+        Timber.w("Update formIndex");
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java:258]: I added this to check if the value in livedata is set by us or maybe it's a restored state as I described in my comment https://github.com/getodk/collect/issues/5241#issuecomment-1206333890

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 2]
```diff
@@ -1396,6 +1396,11 @@ public void onScreenChange(Direction direction) {
      */
     @Override
     public void onScreenRefresh() {
+        FormController formController = getFormController();
+        if (formController == null) {
+            Timber.w("FormController in the application class is null, FormController in formEntryViewModel is %s", (formEntryViewModel.isFormControllerSet() ? "not null" : "null"));
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:1400]: Here I want to know if both formControllers are null or maybe there is only something wrong with the one stored in the application class.

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 3]
```diff
@@ -617,6 +617,7 @@ private void loadForm() {
             }
 
             // Not a restart from a screen orientation change (or other).
+            Timber.w("Setting formController to null");
             Collect.getInstance().setFormController(null);
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: I'm wondering if it wouldn't be better to add logs in that `setFormController` method to see the sequence of calls...
Maybe we first set it properly but then to null?

Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: Yeah that sounds good! Feel to make that change

**Review by lognaturel — APPROVED** [review]

**Pull request merged**